echopype/echodata/zarr_combine.py

```diff
@@ -0,0 +1,742 @@
+        }
+
+        # assign default compressor, if one does not exist
+        if "compressor" not in encodings[str(name)]:
+            encodings[str(name)].update(get_zarr_compression(val, COMPRESSION_SETTINGS["zarr"]))
+
+        # set the chunk encoding
+        encodings[str(name)]["chunks"] = chnk_shape
+
+    def _construct_lazy_ds_and_var_info(
+        self, ds_model: xr.Dataset
+    ) -> Tuple[xr.Dataset, List[str], Dict[str, dict]]:
+        """
+        Constructs a lazy Dataset representing the EchoData group
+        Dataset in its final combined form. Additionally, collects
+        all variable and dimension names that are constant across
+        the Datasets to be combined, and collects the encodings for
+        all variables and dimensions that will be written to the
+        zarr store by regions
+
+        Parameters
+        ----------
+        ds_model: xr.Dataset
+            A Dataset that we will model our lazy Dataset after. In practice,
+            this is the first element in the list of Datasets to be combined.
+
+        Returns
+        -------
+        ds: xr.Dataset
+            A lazy Dataset representing the EchoData group Dataset in
+            its final combined form
+        const_names: List[str]
+            The names of all variables and dimensions that are constant
+            (with respect to chunking) across all Datasets to be combined
+        encodings: Dict[str, dict]
+            The encodings for all variables and dimensions that will be
+            written to the zarr store by regions
+
+        Notes
+        -----
+        The sole purpose of the Dataset created is to construct metadata
+        for the zarr store.
+        """
+
+        xr_vars_dict = dict()
+        xr_coords_dict = dict()
+        encodings = dict()
+        const_names = []
+        for name, val in ds_model.variables.items():
+
+            # get all dimensions of val that are also append dimensions
+            append_dims_in_val = set(val.dims).intersection(self.append_dims)
+
+            if not append_dims_in_val:
+
+                # collect the names of all constant variables/dimensions
+                const_names.append(str(name))
+
+            elif name not in ds_model.dims:
+
+                # create lazy DataArray representations corresponding to the variables
+                temp_arr, chnk_shape = self._get_temp_arr(list(val.dims), val.dtype)
+                xr_vars_dict[name] = (val.dims, temp_arr, val.attrs)
+
+                self._set_encodings(encodings, name, val, chnk_shape)
+
+            elif name in self.append_dims:
+
+                # create lazy DataArray for those coordinates that can be appended to
+                temp_arr, chnk_shape = self._get_temp_arr(list(val.dims), val.dtype)
+                xr_coords_dict[name] = (val.dims, temp_arr, val.attrs)
+
+                self._set_encodings(encodings, name, val, chnk_shape)
```

Comment:
Just to make sure I convey what I tried to describe [above under `_set_encodings()`](https://github.com/OSOceanAcoustics/echopype/pull/808/files/4330422c8f90c9f05f9fafba9e0bb709ea9e55fe#r984185663), I am suggesting to have self._set_encodings to return a new dict and add that to `encodings` here, like:
```python
# add new encodings to the everything-in-one encodings dict here
encodings.update(self._set_encodings(name, val, chnk_shape)) 
```

Reply:
Also, just because the first and last line of both of these `elif` are identical, i think the logic here can be:
```python
        for name, val in ds_model.variables.items():

            # get all dimensions of val that are also append dimensions
            append_dims_in_val = set(val.dims).intersection(self.append_dims)

            if not append_dims_in_val:

                # collect the names of all constant variables/dimensions
                const_names.append(str(name))

            else:

                # create lazy DataArray representations corresponding to the variables
                temp_arr, chnk_shape = self._get_temp_arr(list(val.dims), val.dtype)

                if name not in ds_model.dims:
                     xr_vars_dict[name] = (val.dims, temp_arr, val.attrs)
                elif name in self.append_dims:
                     xr_coords_dict[name] = (val.dims, temp_arr, val.attrs)
                
                # [below i just swapped in the suggestion in the comment immediately above]
                # add new encodings to the everything-in-one encodings dict here
                encodings.update(self._set_encodings(name, val, chnk_shape)) 
```

Reply:
Thank you for the more in depth comment. I agree with your suggestions. I will implement them.